### DISCUSSION TITLE
Optimize performance: Image URL links

### DISCUSSION THREAD

**Discussion by nonoumasy** [vercel/next.js#24918] [Help]
Trying to optimize my site performance. Already improved it a lot by following a tip here regarding not loading Youtube embeds.
Now, I would like to improve the performance even more. I'm hovering around the 80s on the Vercel dashboard for both mobile and web. On lighthouse, I'm getting poorer performance especially on mobile(screenshot).

So my main issue is that my site's content has images which are image url links. I'm rethinking this strategy. At first, I thought this would be ideal as the Users adding content would only need to copy+paste image links and I wouldn't have to host the images, etc. However, now that I'm thinking of performance, I might have to ask my Users to actually upload these images so I can host them on a CDN like Firebase storage or Cloudinary and then I can finally use the Image component on Nextjs and hopefully this will improve my performance. 
Based on what I read on lighthouse and the Vercel analytics, a lot of the bottlenecks is connected to images.

I'm wondering if other people have had this issue and what you did or what you would do in my situation. Would you move to a hosted CDN strategy to host images? Based on this, I might have to refactor my app.

Thanks. 

![Screen Shot 2021-05-08 at 2 42 50 PM](https://user-images.githubusercontent.com/6312676/117554310-b273e280-b00b-11eb-830d-53f8ed680212.png)

**Comment by bobarros** [comment]
Why you can't use Next.js Image Algorithm with local files?

Try at least Caesium and some media queries to reduce the size and use correct image dimensions.

I also had a good experience with Cloudinary, especially with videos, and Sirv (other options) which makes my life easier when I need to control file names. 

Take a look at my website, especially this page that has some images. 
I'm giving you a link to page speed with my website URL and a test --> https://bit.ly/2RAp5Gg

My website repository is public, so you can see what I did.
It's a little bit too much and maybe unnecessary.  Some might argue that might be inefficient. 
But it's going to give you some ideas so you can achieve what you want.

**Reply by nonoumasy** [reply]
@bobarros Thanks again. Yes, I think this is what I eventually have to do to improve performance. No one seems to have a solution to optimize image url links with NextJS.

I'm looking into CDN for images. I've used Cloudinary before and leaning towards that. However, I was wondering if firebase storage is also an option since I'm using Firestore, auth, and functions. The user can upload to storage and then use the link generated as the link to provide to the Image component.

Any thoughts on using Firebase storage vs Cloudinary?

**Comment by bobarros** [comment]
Cloudinary is a service that handles most of the hard work optimizing the images. Firebase is a repository, and if you want to optimize, you must do the work or use another tool.

Like this guys teaches here --> https://maxbarry.medium.com/dynamic-on-demand-image-resizing-using-firebase-hosting-and-google-cloud-functions-to-make-a-cheap-d64e8f5805d1

I still don't understand what do you mean by this: "No one seems to have a solution to optimize image url links with NextJS."
